### PR TITLE
Update facebook dependency to 0.5.0

### PR DESCRIPTION
**Pull request by Serchinastico**
### :pushpin: References
* **Issue:** #35 #33 
* **Related pull-requests:** #20 

### :tophat: What is the goal?

* To increment the facebook screenshot testing library version to, at least, `0.5.0`.
* To be able to run screenshot tests in Android emulators with SDK version greater than 22

### How is it being implemented?

* Increase facebook library version to 0.5.0
* Include a library to be able to parse JSON because the new screenshot test library now uses that format to specify meta-information for each test.
* Replace XML parsing by JSON parsing when retrieving the height and width of the output screenshots.
* Increase android emulator versions to SDK 23 in order to test that the new feature is working.
* To lock the facebook screenshot testing library version to 0.5.0 in case the host application is also loading it. We are doing it so that we don't get incompatible versions installed.

### How can it be tested?

1. Run `./createEmulator.sh`. This will create a new emulator with SDK 23
2. Run `./startEmulator.sh`. This will start the newly created emulator
3. Run `./gradlew executeScreenshotTests` and verify that all tests are passing.

### CODE REVIEW

File: shot/src/main/scala/com/karumi/shot/ShotPlugin.scala [hunk 1]
```diff
@@ -140,9 +140,11 @@ class ShotPlugin extends Plugin[Project] {
...
           facebookDependencyHasBeenAdded |= config.getAllDependencies
             .toArray(new Array[Dependency](0))
-            .exists(dependency =>
-              Config.androidDependencyGroup == dependency.getGroup
-                && Config.androidDependencyName == dependency.getName)
+            .exists(
+              dependency =>
+                Config.androidDependencyGroup == dependency.getGroup
+                  && Config.androidDependencyName == dependency.getName
+                  && Config.androidDependencyVersion == dependency.getVersion)
```
Comment (pedrovgs) [inline shot/src/main/scala/com/karumi/shot/ShotPlugin.scala]: Hum...after reviewing this issue there is something we need to review:

We can check if the version also matches but then we need to update the documentation and consider this change as a **BREAKING CHANGE**. Which will force us to update the documentation and explain the migration in the release description. Here you have the documentation change.

> If you have included in your project a dependency to related to the dexmaker and you are facing this exception: com.android.dx.util.DexException: Multiple dex files define, you can customize how the facebook SDK is added to your project and exclude the dexmaker library as follows:
```
androidTestCompile ('com.facebook.testing.screenshot:core:0.4.2') {
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker'
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker-dx'
  }
```

Should be replaced by:

```
androidTestCompile ('com.facebook.testing.screenshot:core:0.5.0') {
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker'
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker-dx'
  }
```

We should also add a note to remind the user that the version will not be added if the version name matches.

If we don't want to introduce a breaking change, we'd need to update the parser implementation and support both versions and keep this code as is and the new version will work.

The first option seems to be easier to implement, test and maintain, so I'd go for it.

File: core/src/main/scala/com/karumi/shot/xml/ScreenshotsSuiteXmlParser.scala [hunk 2]
```diff
@@ -54,11 +56,15 @@ object ScreenshotsSuiteXmlParser {
...
   def parseScreenshotSize(screenshot: Screenshot,
                           viewHierarchyContent: String): Screenshot = {
-    val xml = XML.loadString(viewHierarchyContent)
-    val screenshotWidth = (xml \ "right" head).text.toInt
-    val screenshotHeight = (xml \ "bottom" head).text.toInt
+    val json = parse(viewHierarchyContent)
+    val JInt(screenshotLeft) = json \ "left"
+    val JInt(screenshotWidth) = json \ "width"
+    val JInt(screenshotTop) = json \ "top"
+    val JInt(screenshotHeight) = json \ "height"
```
Comment (pedrovgs) [inline core/src/main/scala/com/karumi/shot/xml/ScreenshotsSuiteXmlParser.scala]: Are the ``ScreenshotSuiteXmlParser`` tests passing after these changes? I see you are facing some issues with Travis and the emulator set up but we'd need to update the metadata file used in our tests before considering this PR as approved. [Here you have the test suit you should review.](https://github.com/Karumi/Shot/blob/master/core/src/test/scala/com/karumi/shot/xml/ScreenshotsSuiteXmlParserSpec.scala#L20)

Comment (tonilopezmr) [reply core/src/main/scala/com/karumi/shot/xml/ScreenshotsSuiteXmlParser.scala]: We have resolved it, this PR still under development, now, we have some issues with travis.

File: shot/src/main/scala/com/karumi/shot/ShotPlugin.scala (hunk 1 above)
Comment (Serchinastico) [reply shot/src/main/scala/com/karumi/shot/ShotPlugin.scala]: I'm including a more detailed comment in the README asking the users to use v0.5.0 of the facebook library if they want to use them both. Thanks pedro 👍

Comment (pedrovgs) [reply shot/src/main/scala/com/karumi/shot/ShotPlugin.scala]: You are welcome. Let me know when ready and I'll review the final version of this PR. @tonilopezmr would you like to publish the next release with me?

Comment (tonilopezmr) [reply shot/src/main/scala/com/karumi/shot/ShotPlugin.scala]: I want to push the button!

**Comment by tonilopezmr** [comment]
https://github.com/travis-ci/travis-ci/issues/8188
https://travis-ci.org/AllanWang/KAU/jobs/325908661/config

This can help you, maybe.

**Comment by pedrovgs** [comment]
@Serchinastico @tonilopezmr I've just updated this PR replacing Travis CI with Bitrise. I've replaced the old emulator with a new Nexus 5X running Android 25 and updated the create and start emulator scripts. The build is running on my account and as soon as @davideme invites me to our bitrise organization I'll move it there. Thanks to you we are ready to release Shot 2.0.0 I really appreciate your collaboration, time and effort making this project better. Thank you so much 😃

**Pull request merged**